+++
title = "Microbial community analysis by single-amplicon high-throughput next generation sequencing: data analysisâ€“from raw output to ecology"
date = 2016-11-01T00:00:00
draft = false

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Alex J Dumbrell", "Robert MW Ferguson", "**Dave R Clark**"]

# Publication type.
# Legend:
# 0 = Uncategorized
# 1 = Conference paper
# 2 = Journal article
# 3 = Manuscript
# 4 = Report
# 5 = Book
# 6 = Book section
publication_types = ["6"]

# Publication name and optional abbreviated version.
publication = "In *Hydrocarbon and Lipid Microbiology Protocols*."
# publication_short = ""

# Abstract and optional shortened version.
abstract = "Quantifying the functional and taxonomic diversity of microbial assemblages is essential to understanding almost all aspects of microbial ecology. In recent years, the advent of Next Generation Sequencing (NGS) technology has accelerated this process. It is now common practice to target and amplify phylogenetic and/or functional marker genes, and use NGS approaches to characterise their diversity across multiple samples. However, all NGS approaches contain inherent methodological biases, producing both the high-quality data required by researchers, and in addition, erroneous sequences and noise. Careful bioinformatic analysis of NGS data is therefore required to quality filter and process sequences in order to avoid misleading inferences from artifactual results. A similar consideration must also be given to any downstream statistical analysis, as an incorrect choice of approach can also produce false conclusions. These various analytical steps and considerations can appear daunting to the uninitiated and may be perceived as a hurdle to completing research. In this chapter, we aim to provide the methods and guidance required to overcome this hurdle, and impart the skills required for a novice bioinformatician to produce a basic analysis of their NGS amplicon data. We focus on data produced using two common NGS technologies, the historically more widely used 454-pyrosequencer, and the currently more widely used Illumina platform. We cover methods for quality filtering and denoising data, picking Operational Taxonomic Units (OTUs), assigning taxonomy to sequences and basic statistical analyses required for hypothesis testing. Implementation of these methods is demonstrated for both of two commonly used pipelines (QIIME and mothur), and additional stand-alone packages (including R), providing the reader with maximum flexibility when analysing their data."

# Is this a selected publication? (true/false)
selected = true

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = ["16S rRNA", "454-Pyrosequencing", "Amplicon", "Bioinformatics", "BioLinux", "Illumina", "Microbial ecology", "Mothur", "Next generation sequencing", "QIIME"]

# Links (optional).
url_pdf = "https://link.springer.com/protocol/10.1007/8623_2016_228"

# Custom links (optional).
#   Uncomment line below to enable. For multiple links, use the form `[{...}, {...}, {...}]`.
# url_custom = [{name = "Custom Link", url = "http://example.org"}]

# Digital Object Identifier (DOI)
doi = ""

# Does this page contain LaTeX math? (true/false)
math = false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
[image]
  # Caption (optional)
  caption = "Image credit: [**Unsplash**](https://unsplash.com/photos/pLCdAaMFLTE)"

  # Focal point (optional)
  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
  focal_point = ""
+++
